### ISSUE TITLE
ProductNotFound in 546, 547, 548 workflows

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#33578]
Workflows 546, 547, 548 step 2, 3, 3 (respectively) has been failing since CMSSW_12_0_X_2021-04-28-1100 with
```
%MSG
----- Begin Fatal Exception 28-Apr-2021 15:37:20 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 1
   [1] Running path 'nanoAOD_step'
   [2] Calling method for module GenVisTauProducer/'genVisTaus'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::GenJet>
Looking for module label: tauGenJetsSelectorAllHadronsForNano
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
%MSG-w MemoryCheck:  JetFlavourClustering:genJetAK8FlavourAssociation  28-Apr-2021 15:37:20 C
```
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc900/CMSSW_12_0_X_2021-04-28-1100/pyRelValMatrixLogs/run/546.0_DYToLL_M-50_13TeV_pythia8+DYToLL_M-50_13TeV_pythia8+NANOGENFromGen/step2_DYToLL_M-50_13TeV_pythia8+DYToLL_M-50_13TeV_pythia8+NANOGENFromGen.log#/

**Comment by makortel** [comment]
assign generators, xpog

**Comment by makortel** [comment]
https://github.com/cms-sw/cmssw/pull/33522 (merged in that IB) seems like a plausible cause

**Comment by mariadalfonso** [comment]
@mbluj 
can you help here ?

**Comment by kdlong** [comment]
This looks like the likely culprit. Best would be to divide the tauMC statement into the genOnly pieces so the actual sequence can be added to NanoGen. Something like:

tauMCgen = cms.Sequence(tauGenJetsForNano + tauGenJetsSelectorAllHadronsForNano + genVisTaus + genVisTauTable)
tauMC = cms.Sequence(tauMCGen + tausMCMatchLepTauForTable + tausMCMatchHadTauForTable + tauMCTable)

then schedule tauMCgen in the NanoGen sequence. If someone else can actually implement this (@dteague ?) that would be great.

**Comment by mbluj** [comment]
I think I can take it.
@kdlong, thanks for suggestion, I'll follow it.

**Comment by mbluj** [comment]
Fixes implemented in PR #33586 (master) and #33590 (106X).

**Comment by mariadalfonso** [comment]
+xpog
